server/src/main/java/org/apache/iotdb/db/metadata/logfile/MLogWriter.java

```diff
@@ -0,0 +1,340 @@
+    } catch (BufferOverflowException e) {
+      logger.error("MLog {} BufferOverflow !", plan.getOperatorType(), e);
+      mlogBuffer.reset();
+      sync();
+      plan.serialize(mlogBuffer);
+    }
+    logNum ++;
+  }
+
+  public void createTimeseries(CreateTimeSeriesPlan plan, long offset) throws IOException {
+    try {
+      putLog(plan);
+      ChangeTagOffsetPlan changeTagOffsetPlan = new ChangeTagOffsetPlan(plan.getPath(), offset);
+      putLog(changeTagOffsetPlan);
+    } catch (BufferOverflowException e) {
+      throw new IOException(
+        "Log cannot fit into buffer, please increase mlog_buffer_size", e);
+    }
+  }
+
+  public void deleteTimeseries(DeleteTimeSeriesPlan deleteTimeSeriesPlan) throws IOException {
+    try {
+      putLog(deleteTimeSeriesPlan);
+    } catch (BufferOverflowException e) {
+      throw new IOException(
+        "Log cannot fit into buffer, please increase mlog_buffer_size", e);
+    }
+  }
+
+  public void setStorageGroup(PartialPath storageGroup) throws IOException {
+    try {
+      SetStorageGroupPlan plan = new SetStorageGroupPlan(storageGroup);
+      putLog(plan);
+    } catch (BufferOverflowException e) {
+      throw new IOException(
+        "Log cannot fit into buffer, please increase mlog_buffer_size", e);
+    }
+  }
+
+  public void deleteStorageGroup(PartialPath storageGroup) throws IOException {
+    try {
+      DeleteStorageGroupPlan plan = new DeleteStorageGroupPlan(Collections.singletonList(storageGroup));
+      putLog(plan);
+    } catch (BufferOverflowException e) {
+      throw new IOException(
+        "Log cannot fit into buffer, please increase mlog_buffer_size", e);
+    }
+  }
+
+  public void setTTL(PartialPath storageGroup, long ttl) throws IOException {
+    try {
+      SetTTLPlan plan = new SetTTLPlan(storageGroup, ttl);
+      putLog(plan);
+    } catch (BufferOverflowException e) {
+      throw new IOException(
+        "Log cannot fit into buffer, please increase mlog_buffer_size", e);
+    }
+  }
+
+  public void changeOffset(PartialPath path, long offset) throws IOException {
+    try {
+      ChangeTagOffsetPlan plan = new ChangeTagOffsetPlan(path, offset);
+      putLog(plan);
+    } catch (BufferOverflowException e) {
+      throw new IOException(
+        "Log cannot fit into buffer, please increase mlog_buffer_size", e);
+    }
+  }
+
+  public void changeAlias(PartialPath path, String alias) throws IOException {
+    try {
+      ChangeAliasPlan plan = new ChangeAliasPlan(path, alias);
+      putLog(plan);
+    } catch (BufferOverflowException e) {
+      throw new IOException(
+        "Log cannot fit into buffer, please increase mlog_buffer_size", e);
+    }
+  }
+
+  public void serializeMNode(MNode node) throws IOException {
+    try {
+      int childSize = 0;
+      if (node.getChildren() != null) {
+        childSize = node.getChildren().size();
+      }
+      MNodePlan plan = new MNodePlan(node.getName(), childSize);
+      putLog(plan);
```

Comment:
Hi, may I ask why we put operations like `serializeMNode` to log?

Reply:
It's snapshot file, not mlog.bin, I just serialize the snapshot and mlog by the same logwriter.

Reply:
Ah... I understand, it used to be the snapshot file which serializes the MTree. @qiaojialin Do you think it is acceptable?